### PR TITLE
stm32 spi bus driver

### PR DESCRIPTION
**Pull request by kasjer**
This adds SPI bus driver that utilizes DMA.

Code uses ST HAL drivers for all MCUs.

### CODE REVIEW

**Review by utzig — APPROVED** [review]
LGTM

File: hw/bus/drivers/spi_stm32/include/bus/drivers/spi_stm32.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#endif
+
+/**
+ * Initialize os_dev as SPI bus device using spi_hal driver
+ *
+ * This can be passed as a parameter to os_dev_create() when creating os_dev
+ * object for SPI device, however it's recommended to create devices using helper
+ * like bus_spi_hal_dev_create().
+ *
+ * @param node  Node device object
```
[the remark this answers is not in the document]
Comment (kasjer) [reply hw/bus/drivers/spi_stm32/include/bus/drivers/spi_stm32.h]: fixed

**Comment by utzig** [comment]
@kasjer Please rebase.

**Comment by kasjer** [comment]
rebased as requested

**Pull request merged**